bridge-token-factory/src/lib.rs

```diff
@@ -394,12 +348,36 @@ impl BridgeTokenFactory {
     }
 }
 
+#[near_bindgen]
+impl FungibleTokenReceiver for BridgeTokenFactory {
+    fn ft_on_transfer(
+        &mut self,
+        _sender_id: ValidAccountId,
+        amount: U128,
+        msg: String,
+    ) -> PromiseOrValue<U128> {
+        assert!(false, "Native FT on Ethereum is disabled");
+        let recipient = validate_eth_address(msg);
+        env::log(
+            &ResultType::Lock {
+                token: env::predecessor_account_id(),
+                amount: amount.into(),
+                recipient,
+            }
+            .try_to_vec()
+            .unwrap(),
+        );
+        PromiseOrValue::Value(amount)
```

Comment:
Should this be `0` given that we are locking all those tokens on this contract?

Reply:
You are right, should be 0